src/python/pants/jvm/goals/lockfile.py

```diff
@@ -159,11 +164,51 @@ async def setup_user_lockfile_requests(
     return UserGenerateLockfiles(jvm_lockfile_requests)
 
 
+@dataclass(frozen=True)
+class JvmSyntheticLockfileTargetsRequest(SyntheticTargetsRequest):
+    """Register the type used to create synthetic targets for JVM lockfiles.
+
+    As the paths for all lockfiles are known up-front, we set the `path` field to
+    `SyntheticTargetsRequest.SINGLE_REQUEST_FOR_ALL_TARGETS` so that we get a single request for all
+    our synthetic targets rather than one request per directory.
+    """
+
+    path: str = SyntheticTargetsRequest.SINGLE_REQUEST_FOR_ALL_TARGETS
+
+
+@rule
+async def jvm_lockfile_synthetic_targets(
+    request: JvmSyntheticLockfileTargetsRequest,
+    jvm_subsystem: JvmSubsystem,
+) -> SyntheticAddressMaps:
+    if not jvm_subsystem.enable_lockfile_targets:
+        return SyntheticAddressMaps()
+
```

Comment:
To make this target useful, something would need to depend on it, right? Are some additional `@rules` missing to inject a dependency on these targets?

Reply:
Yes, this PR is just a skeleton thus far (as I recall, a bit vague on the details atm) as I'm not at all familiar with the Java backend (or Java/Scala in general) so are very unsure where to go with this.

Started this to kick off some activity to get on par with the Python functionality wrgt dependencies on generated lockfiles.